app/src/main/java/com/example/taskyy/data/remote/interceptors/ApiKeyInterceptor.kt

```diff
@@ -13,6 +15,7 @@ class ApiKeyInterceptor(
                 .request()
                 .newBuilder()
                 .addHeader("x-api-key", apiKey)
+                .addHeader("token", userPreferences.getUserToken())
```

Comment:
If this interceptor apparently does more than just adding the API key, I'd also communicate that with its name. You could just call it `AuthInterceptor` instead